File: Clases python/Clases/Automovil.py
from Clases.Vehiculo import*
class Automovil(Vehiculo):
    def __init__(self, fecha_fabricacion, VIN_Chasis, VINMotor, marca, modelo, precio):
        super().__init__(fecha_fabricacion, VIN_Chasis, VINMotor) 
        self.marca = marca
        self.modelo = modelo
        self.precio = precio
    @property
    def marca(self):
        return self._marca

    @marca.setter
    def marca(self, valor):
        self._marca = valor
    
    @property
    def modelo(self):
        return self._modelo

    @modelo.setter
    def modelo(self, valor):
        self._modelo = valor
    
    @property
    def precio(self):
        return self._precio

    @precio.setter
    def precio(self, valor):
        self._precio = valor
    
    def mostrar_datos(self):
        print(f"Datos del Automóvil:")
        print(f"Marca: {self.marca}")
        print(f"Modelo: {self.modelo}")
        print(f"Precio: ${self.precio:,.2f}")
        print(f"Fecha de Fabricación: {self.fecha_fabricacion}")
        print(f"VIN Chasis: {self.VIN_Chasis}")
        print(f"VIN Motor: {self.VINMotor}")

File: Herencia/Clases/src/clases/Automovil.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clases;

/**
 *
 * @author ESTUDIANTES
 */

public class Automovil extends Vehiculo {

    private String marca;
    private String modelo;
    private double precio;

    public Automovil(String fechaFabricacion, String VINChasis, String VINMotor, String marca, String modelo, double precio) {
        
        super(fechaFabricacion, VINChasis, VINMotor);
        this.marca = marca;
        this.modelo = modelo;
        this.precio = precio;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public void mostrarDatos() {
        System.out.println("Datos del Automóvil:");
        System.out.println("Marca: " + getMarca());
        System.out.println("Modelo: " + getModelo());
        System.out.println("Precio: $" + String.format("%.2f", getPrecio()));
        System.out.println("Fecha de Fabricación: " + getFechaFabricacion());
        System.out.println("VIN Chasis: " + getVINChasis());
        System.out.println("VIN Motor: " + getVINMotor());
    }
}
